coral-dragaction/src/scripts/DragAction.js

```diff
@@ -682,6 +687,45 @@ class DragAction {
     }
   }
 
+  /** @private */
+  _dragOnKey(event) {
+    switch (event.code) {
+      case 'Space':
+        this._dragEvents.dispatch('coral-dragaction:dragonkeyspace', {
+          detail: {
+            dragElement: this._dragElement
+          }
+        });
+        break;
+      case 'ArrowDown':
+        this._dragEvents.dispatch('coral-dragaction:dragoveronkeyarrowdown', {
+          detail: {
+            dragElement: this._dragElement
+          }
+        });
+        break;
+      case 'ArrowUp':
+        this._dragEvents.dispatch('coral-dragaction:dragoveronkeyarrowup', {
+          detail: {
+            dragElement: this._dragElement
+          }
+        });
+        break;
+      case 'Enter':
+        this._dragOnKeyEnd(event);
+        break;
+    }
+  }  
+
+  /** @private */
+  _dragOnKeyEnd(event) {
+    this._dragEvents.dispatch('coral-dragaction:dragendonkey', {
+      detail: {
+        dragElement: this._dragElement
+      }
+    });
```

Comment:
Remove the `focusout` event handler when the keyboard drag and drop operation ends.
```suggestion
    this._dragEvents.dispatch('coral-dragaction:dragendonkey', {
      detail: {
        dragElement: this._dragElement
      }
    });
    events.off(`focusout.DragAction${this._id}`, this._dragOnKeyEnd);
```